validation/validate.py

```diff
@@ -19,17 +26,29 @@ def validate(collection_path):
             try:
                 in_data = yaml.safe_load(stream)
             except yaml.YAMLError as exc:
-                print(exc)
-
-        # validate it
-        # try:
-        #     CollectionSchema().load(in_data)
-        # except ValidationError as err:
-        #     pprint(err.messages)
-
-        print(f"validating {collection}")
-        parsed = CollectionSchema.parse_obj(in_data)
-        print(parsed)
+                yaml_error_list.append((str(exc.problem) + str(exc.problem_mark)))
+                continue
+
+        # process the Collection
+        try:
+            CollectionSchema.parse_obj(in_data)
+        except ValidationError as err:
+            schema_error_list.append(str(err) + ' in ' + collection)
```

Comment:
The reason that I have the `yaml.safe_load` snippet first and then the `CollectionSchema.parse_bj` snippet later is that if the code encounters the yaml error, it just moves on to the next collection in the for loop after appending the necessary info in `yaml_error_list`. The only case that `parse_obj` is called is after ensuring that `yaml.safe_load` did not error.

Reply:
If I understand your concern correctly, I believe the above snippet should also be doing the same.

The above snippet would only go on to the `CollectionSchema.parse_obj(in_data)` if the `safe_load(stream)` were executed successfully. Otherwise, if it encountered an exception, it would go on to the `except` block and, subsequently, process the following collection.